src/SIL.XForge.Scripture/ClientApp/src/app/translate/editor/editor.component.spec.ts

```diff
@@ -3469,8 +3495,10 @@ class TestEnvironment {
   }
 
   isNoteIconHighlighted(threadId: string): boolean {
+    const note = this.targetTextEditor.querySelector(`usx-segment display-note[data-thread-id=${threadId}]`);
+    expect(note).withContext('note thread highlight').not.toBeNull();
     const thread: HTMLElement | null = this.targetTextEditor.querySelector(
-      `usx-segment display-note[data-thread-id="${threadId}"].note-thread-highlight`
+      `usx-segment display-note[data-thread-id=${threadId}].note-thread-highlight`
     );
     return thread != null;
```

Comment:
I think this would be a much better test if you selected the thread and then checked whether the thread has the `note-thread-highlight` class. As it exists now, if your selector was wrong you would get a negative result, and getting selectors wrong is really easy (especially if someone comes and edits the tests later.)

Maybe something like this:
``` suggestion
    const thread: HTMLElement | null = this.targetTextEditor.querySelector(
      `usx-segment display-note[data-thread-id=${threadId}]`
    );
    expect(thread).withContext('note thread highlight').not.toBeNull();
    return thread!.classList.contains('note-thread-highlight');
```

Reply:
Great thinking. I've made these changes.